audio-echo/app/build.gradle

```diff
@@ -6,22 +6,18 @@ android {
     defaultConfig {
         applicationId 'com.google.sample.echo'
         /*
-           To run on earlier version of Android than android-21, do:
+           To run on earlier version of Android other than android-23, do:
              *) set this minSDKVersion and cmake's ANDROID_PLATFORM to your version
-             *) set ANDROID_STL to c++_static for some very earlier version android.
+             *) set ANDROID_STL to c++_static for earlier Android API levels
```

Comment:
Is this a case of `NativeActivity` being broken pre M? There's not enough context in this diff to be very helpful.

Reply:
no, this does not use the NativeActivity, directly using JNI.  NativeActivity thing is about Android-24 -- pre-N does not load dependency shared lib, N and after NativeActivity does.

For (very )early version of the apps I believe static is flavor recommended. Now it is shared.

Reply:
> For (very )early version of the apps I believe static is flavor recommended. Now it is shared.

What does API level have to do with this (aside from the bug with `NativeActivity`)? As far as I know the guidance is to use the static STL unless you have multiple shared libraries in your app, in which case you should use the shared STL.